uvicorn/middleware/proxy_headers.py

```diff
@@ -70,15 +150,22 @@ async def __call__(
                         scope["scheme"] = x_forwarded_proto
 
                 if b"x-forwarded-for" in headers:
-                    # Determine the client address from the last trusted IP in the
-                    # X-Forwarded-For header. We've lost the connecting client's port
-                    # information by now, so only include the host.
                     x_forwarded_for = headers[b"x-forwarded-for"].decode("latin1")
-                    x_forwarded_for_hosts = [
-                        item.strip() for item in x_forwarded_for.split(",")
-                    ]
-                    host = self.get_trusted_client_host(x_forwarded_for_hosts)
-                    port = 0
-                    scope["client"] = (host, port)  # type: ignore[arg-type]
+                    host = self.trusted_hosts.get_trusted_client_host(x_forwarded_for)
+
+                    if host:
+                        # If the x-forwarded-for header is empty then host is None or
+                        # an empty string.
+                        # Only set the client if we actually got something usable.
+                        # See: https://github.com/encode/uvicorn/issues/1068
+
+                        # Unless we can relaibly use x-forwarded-port (see below) then
+                        # we will not have any port information so set it to 0.
+                        port = 0
+                        scope["client"] = (host, port)
+
+                    # if b"x-forwarded-port" in headers:
+                    #     ...
+                    # https://github.com/encode/uvicorn/issues/1974
```

Comment:
- I've reopened https://github.com/encode/uvicorn/issues/1974, and I'll accept a follow-up PR.
```suggestion
                        # We've lost the connecting client's port information by now,
                        # so only include the host.
                        port = 0
                        scope["client"] = (host, port)
```